Blog/urls.py

```diff
@@ -7,13 +7,13 @@
 urlpatterns = [
     path('', views.index, name="index"),
     path('posts/', views.PostListView.as_view(), name="posts"),
-    path('posts/<int:pk>', views.post_detail_view, name="post-detail"),
+    path('posts/view/<slug:slug>', views.post_detail_view, name="post-detail"),
```

Comment:
There is a problem:
If the pattern is missing the segment `view` (i.e. the pattern would be `posts/<slug:slug>`), django will always try to display a post, even if the part behind `posts/` is not a slug.

This means, the form to add an own post is not accessible anymore. The form can be reached under `posts/add`. since django assumes that `add` is a slug which corresponds to a post. Instead of displaying the `Add Post` form, it tries to display the (maybe non existing) post in a `post-detail` view.

Reply:
Excellent analysation.

Could this be solved by putting 

`path('posts/add', views.PostCreate.as_view(), name="post-add"),`

before

`path('posts/<slug:slug>', views.post_detail_view, name="post-detail"),`

?

Reply:
It would work; However, with this solution there comes another problem.
If an user now adds a post with the title "add", django won't recognize "add" as a slug and therefore redirects the user to the "Add Post" page. 

Means, if you click on the details page of the post with the title `add`, you will land on the `Add a new Post`-page.

two solutions: Either move `Posts/add` to another URL (like `/add/post`) or add the `view` segment to the URL (as implemented, I.e. `Posts/view/<slug:slug>`).

Reply:
Very good point 👍 

So we just leave the `view` segment here and discuss this later, so you can move on with this Pull Request.

Reply:
As discussed:

- Post read view url:  `posts/<slug>`
- Post create view url: `post/add`